### PR TITLE
Bi 7516 add paging to dissolved best match

### PR DESCRIPTION
**Pull request by ethatcher83**
Added pagination to "company name when it was dissolved" best match results page

Resolves: BI-7516

### CODE REVIEW

File: src/client/apiclient.ts [hunk 1]
```diff
@@ -21,10 +21,10 @@ export const getCompanies =
     };
 
 export const getDissolvedCompanies =
-async (apiKey: string, companyName: string, requestId, searchType: string): Promise<DissolvedCompaniesResource> => {
+async (apiKey: string, companyName: string, requestId, searchType: string, page: number): Promise<DissolvedCompaniesResource> => {
     const api = createApiClient(apiKey, undefined, API_URL);
     const companiesResource: Resource<DissolvedCompaniesResource> =
-        await api.dissolvedSearch.getCompanies(companyName, requestId, searchType);
+        await api.dissolvedSearch.getCompanies(companyName, requestId, searchType, page * DISSOLVED_SEARCH_NUMBER_OF_RESULTS - 20);
```
Comment (mikecsmith) [inline src/client/apiclient.ts:27]: I'd be tempted to define the calculation here as a constant - also - is the `-20` just `- DISSOLVED_SEARCH_NUMBER_OF_RESULTS`?

File: src/views/components/pagination.html [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            <a class="page" id="previous-page" href="{{partialHref}}&page={{page - 1}}">
+                Previous
+            </a>
+        </li>
+        {% endif %}
+        {% if numberOfPages < 10 %}
+            {% for i in range(1, numberOfPages + 1) -%}
+                {% if (i == page) %}
+                    <li>
+                        <a id="page-{{ i }}" class="active" href="{{partialHref}}&page={{i}}">{{ i }}</a>
```
Comment (mikecsmith) [inline src/views/components/pagination.html:14]: Should this be an anchor tag - or should it just be an `<li>`?

File: src/views/components/pagination.html [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                    </li>
+                {% endif %}
+            {%- endfor %}
+        {% endif %}
+
+        {% if numberOfPages >= 10 and page < 5 %}
+            {% for i in range(1, 11) -%}
+                {% if (i == page) %}
+                    <li>
+                        <a id="page-{{ i }}" class="active" href="{{partialHref}}&page={{i}}">{{ i }}</a>
```
Comment (mikecsmith) [inline src/views/components/pagination.html:28]: Same as above

File: src/views/components/pagination.html [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                    </li>
+                {% endif %}
+            {%- endfor %}
+        {% endif %}
+
+        {% if (numberOfPages > 10 and page >= 5) and (numberOfPages > 10 and page < numberOfPages - 6) %}
+            {% for i in range( page -4, page + 6) -%}
+                {% if (i == page) %}
+                    <li>
+                        <a id="page-{{ i }}" class="active" href="{{partialHref}}&page={{i}}">{{ i }}</a>
```
Comment (mikecsmith) [inline src/views/components/pagination.html:42]: Same as above

File: src/views/components/pagination.html [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                    </li>
+                {% endif %}
+            {%- endfor %}
+        {% endif %}
+
+        {% if page > 10 and page >= numberOfPages - 6 %}
+            {% for i in range(numberOfPages -9 , numberOfPages + 1) -%}
+                {% if (i == page) %}
+                <li>
+                    <a id="page-{{ i }}" class="active" href="{{partialHref}}&page={{i}}">{{ i }}</a>
```
Comment (mikecsmith) [inline src/views/components/pagination.html:56]: Same as above

File: src/views/components/pagination.html [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                    </li>
+                {% endif %}
+            {%- endfor %}
+        {% endif %}
+
+        {% if (numberOfPages - page)  <= 10 %}
+            {% for i in range(numberOfPages, numberOfPages - 10) -%}
+                {% if (i == page) %}
+                <li>
+                    <a id="page-{{ i }}" class="active" href="{{partialHref}}&page={{i}}">{{ i }}</a>
```
Comment (mikecsmith) [inline src/views/components/pagination.html:70]: Same as above

**Review by katherine-axten — APPROVED** [review]

**Pull request merged**